### ISSUE TITLE
[Bug] Updating to v1.3 causes "Unexpected error fetching TrueNAS data"

### ISSUE THREAD

**Issue by thepurpleplatypus** [tomaae/homeassistant-truenas#186]
## Describe the issue
Since updating to v1.3 of this extension, I am getting "Unexpected error fetching TrueNAS data".  This was working fine with the previous build (v1.3b2 I think?)

## How to reproduce the issue
Update the extension to v1.3 and restart HA

## Expected behavior
It should have kept working as before

## Software versions

 - Home Assistant version: 2024.11.1
 - TrueNAS integration version: v1.3
 - TrueNAS version: Core TrueNAS-13.0-U6.2

## Diagnostics data

  Unexpected error fetching TrueNAS data
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 382, in _async_refresh
    self.data = await self._async_update_data()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/truenas/coordinator.py", line 95, in _async_update_data
    await self.hass.async_add_executor_job(self.get_systeminfo)
  File "/usr/local/lib/python3.12/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/truenas/coordinator.py", line 221, in get_systeminfo
    self._version_minor = int(
                          ^^^^
ValueError: invalid literal for int() with base 10: '0-U6'

## Additional context
I think the issue may be due to the changes from commit fcdc800 but I've not been able to validate this yet.

**Comment by Rahi0312** [comment]
Hi having the same problem with TueNAS 13.3 release:

Unexpected error fetching TrueNAS data
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 382, in _async_refresh
    self.data = await self._async_update_data()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/truenas/coordinator.py", line 95, in _async_update_data
    await self.hass.async_add_executor_job(self.get_systeminfo)
  File "/usr/local/lib/python3.12/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/truenas/coordinator.py", line 221, in get_systeminfo
    self._version_minor = int(
                          ^^^^
ValueError: invalid literal for int() with base 10: '3-RELEASE'

**Comment by gluckb** [comment]
hi,
this worked for me:
open "/config/custom_components/truenas/coordinator.py" with your file editor
insert line 226:
`.split("-")[0]`
before line:
`.split(".")[1]`

**Comment by raidflex** [comment]
Confirmed added the above line fixed it as well for me.

**Comment by tomaae** [comment]
Fix is just above your post. It was already released as 1.3.1

**Comment by Daviderusso93** [comment]
Hi at all.
I have the same issue today (19 December 2024).

![image](https://github.com/user-attachments/assets/3ee425e1-8c62-4ae5-a859-22f39a45c3aa)

`Questo errore ha avuto origine da un'integrazione personalizzata.

```
Registratore: custom_components.truenas.coordinator
Fonte: helpers/update_coordinator.py:379
Integrazione: TrueNAS (documentazione, problemi)
Prima occorrenza: 09:34:04 (43 occorrenze)
Ultimo accesso: 14:08:06

Unexpected error fetching TrueNAS data
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 379, in _async_refresh
    self.data = await self._async_update_data()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/truenas/coordinator.py", line 97, in _async_update_data
    await self.hass.async_add_executor_job(self.get_systemstats)
  File "/usr/local/lib/python3.13/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/truenas/coordinator.py", line 445, in get_systemstats
    max(tmp_graph[i]["aggregations"]["mean"].values()), 2
    ~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: max() iterable argument is empty`
```

Home Assistant version: 2024.12.24
TrueNAS Scale version: 24.10.1 (EletricEel-24.10.1)

Both systems are on VM (Hyper-V).
They are connected with two different virtual connectors, which work on two different LAN cards.

From the HA VM I can ping the TrueNAS VM and it works the other way around too.
I can connect it with the API Key generated by TrueNAS, but I always get this error.

I saw the proposed solution by modifying the value of line 226 of the coordinator.py file (as was done with the old version 1.3) but it didn't work.
I've tried both 1.3 and 1.3.1, but neither seems to work.

Any ideas?
